### PR TITLE
Prometheus

### LINKED ISSUE #6: Prometheus exporter & Prometheus server setup
We need to get monitoring working on the new Proxmox setup, so someone needs to write a playbook that sets up Prometheus exporters on every server.

For now, they can create a new test vm with a new `create-*.yml` playbook. They will then need to target this vm with a similar `provision-.yml` playbook. This should install the following Prometheus exporters (preferably) in Docker:

* cadvisor
* node_exporter
* ipmi_exporter (maybe?)

These exporters should be reverse proxified by expexp_exporter

There is an existing implementation of this in `roles/prometheus-exporters` but I do not know how well it works/if at all. I think I moved it originally from a playbook to a role

Whoever completes this issue will also need to setup a Prometheus server (probably on the docker swarm in `provision-docker-swarm-services.yml`) that collects the metrics exported by the testvm

Once we have the test vm working, we can call the playbook on all our vms and setup metrics for everything

### PR DESCRIPTION
**Pull request by jac**
This closes #6 

This will all soon be documented on the wiki but a quick overview for the pr

### Prometheus Exporters
_provision-prometheus-exporters.yml_ targets the group "prometheus_base"

"prometheus_base" is required for all hosts which are exporters as it installs "exporter exporter" which is used to proxy all other exporters. All other groups are optional
| group name | exporters installed |
| ----------------- | ------------------------ |
| prometheus_base | node_exporter, exporter exporter |
| prometheus_docker | cadvisor |
| prometheus_apache | apache_exporter |
| prometheus_ipmi | ipmi_exporter |

Note: "exporter exporter" will also proxy minecraft metrics on games.vm.netsoc.co. The minecraft exporter is installed as a bukkit plugin

### Prometheus Server
Sidenote: Initially the prometheus server was going to be put on swarm but prom kept crashing as it doesn't work well with nfs 🤷

The prometheus server is installed on the infra databases server. It can be accessed internally without auth at prometheus.databases.vm.netsoc.co on the infra vlan or externally at prometheus.netsoc.dev using basic auth (reverse proxied by traefik)

It uses the groups to configure scrape targets. It is also configured to scrape minecraft metrics as well as check if netsoc websites are online using blackbox_exporter

### CODE REVIEW

File: roles/prometheus-exporter/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      - "/var/run:/var/run:rw"
+      - "/sys:/sys:ro"
+      - "/var/lib/docker/:/var/lib/docker:ro"
+      - "/dev/disk/:/dev/disk:ro"
+
+- name: IPMI Exporter
+  when: "'prometheus_ipmi' in group_names"
+  docker_container:
+    name: ipmi_exporter
+    image: "mloc6/ipmi_exporter"
```
Comment (ocanty) [inline roles/prometheus-exporter/tasks/main.yml:31]: @ericm push to dockerhub pls

File: roles/prometheus-server/tasks/main.yml [hunk 2]
```diff
@@ -1,110 +1,91 @@
 ---
-- name: "Create /netsoc-neo/docker-data/prometheus directory"
+- name: "Create /mnt/data-disk/prometheus directory"
   become: yes
   file:
-    path: /netsoc-neo/docker-data/prometheus
-    mode: 0770 
+    path: /mnt/data-disk/prometheus
+    mode: 0777
```
Comment (ocanty) [inline roles/prometheus-server/tasks/main.yml:6]: 0770

File: roles/prometheus-server/tasks/main.yml [hunk 3]
```diff
@@ -1,110 +1,91 @@
...
     state: directory
 
-- name: "Create /netsoc-neo/docker-data/prometheus/config directory"
+- name: "Create /mnt/data-disk/prometheus/config directory"
   become: yes
   file:
-    path: /netsoc-neo/docker-data/prometheus/config
-    mode: 0770 
+    path: /mnt/data-disk/prometheus/config
+    mode: 0777
```
Comment (ocanty) [inline roles/prometheus-server/tasks/main.yml:15]: 0770

File: roles/prometheus-server/tasks/main.yml [hunk 4]
```diff
@@ -1,110 +1,91 @@
...
     state: directory
 
-- name: "Create /netsoc-neo/docker-data/blackbox_exporter directory"
+- name: "Create /mnt/data-disk/prometheus/data directory"
   become: yes
   file:
-    path: /netsoc-neo/docker-data/blackbox_exporter
-    mode: 0770 
+    path: /mnt/data-disk/prometheus/data
+    mode: 0777
```
Comment (ocanty) [inline roles/prometheus-server/tasks/main.yml:24]: 0770

File: vars/traefik.yml [hunk 5]
```diff
@@ -1 +1,5 @@
-traefik_basic_auth: "{{ vault_traefik_basic_auth }}"
\ No newline at end of file
+traefik_basic_auth: "{{ vault_traefik_basic_auth }}"
+
+traefik:
```
Comment (ocanty) [inline vars/traefik.yml:3]: you can change this to just `traefik: {{ vault_traefik }}`

Comment (jac) [reply vars/traefik.yml:3]: I did just vault_traefik initially but kind of liked the more explicit way. I can change it to just vault_traefik though

Comment (ocanty) [reply vars/traefik.yml:3]: gonna amalgamate and delete all of vars/*.yml soon so want to keep the line count down

then we can document a copy of vars/secrets.yml in the readme  with none of the passwords showing

**Comment by ocanty** [comment]
lgtm

**Pull request merged**